### PR TITLE
test: add test for needSwitchDB

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 

See: https://github.com/pingcap/ticdc/pull/3264/files#r745243747

### What is changed and how it works?

1. move to better place.
2. add test for it

### Check List 

Tests 

 - Unit test

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note

None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @lance6716 @amyangfei 

Could you please take a look? Thanks!

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -364,6 +354,16 @@ func (s *mysqlSink) execDDL(ctx context.Context, ddl *model.DDLEvent) error {
 	return nil
 }
 
+func needSwitchDB(ddl *model.DDLEvent) bool {
```
Comment (lance6716) [inline cdc/sink/mysql.go:357]: to reduce indent level, we can

if len == 0 {
return false
}
if ddl.Type == timodel.ActionCreateSchema || ddl.Type == timodel.ActionDropSchema {
return false
}
return true

also you can use `switch` to replace last `if`

Comment (hi-rustin) [reply cdc/sink/mysql.go:357]: Done. Thanks!

**Review by lance6716 — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]

**Comment by hi-rustin** [comment]
/hold

Waiting for @amyangfei review.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**